packages/salesforcedx-vscode-core/src/commands/baseDeployRetrieve.ts

```diff
@@ -118,13 +118,15 @@ export abstract class DeployExecutor<T> extends DeployRetrieveExecutor<T> {
   ): Promise<DeployResult | undefined> {
     const projectPath = getRootWorkspacePath();
     const connection = await WorkspaceContext.getInstance().getConnection();
-
     components.projectDirectory = projectPath;
-    const sourceTracking = await SourceTrackingService.createSourceTracking(
-      projectPath,
-      connection
-    );
-    await sourceTracking.ensureLocalTracking();
+    const sourceTrackingEnabled = sfdxCoreSettings.getEnableSourceTrackingForDeployAndRetrieve();
+    if (sourceTrackingEnabled) {
```

Comment:
Adding the source tracking info in deploy executor under the setting if SourceTracking is Enabled.